src/ParseNode.js

```diff
@@ -40,18 +41,83 @@ export type NodeValue<TYPE: NodeType> = $ElementType<ParseNodeTypes, TYPE>;
 
 export type LeftRightDelimType = {|
     type: "leftright",
-    body: ParseNode<*>[],
+    body: AnyParseNode[],
     left: string,
     right: string,
 |};
 
+// ParseNode's corresponding to Symbol `Group`s in symbols.js.
+export type SymbolParseNode =
```

Comment:
This is definitely a step in the right direction, but not it's final form.  That can wait for another PR.